src/librustc/ty/sty.rs

```diff
@@ -862,23 +863,24 @@ impl<'tcx> PolyFnSig<'tcx> {
 pub struct ParamTy {
     pub idx: u32,
     pub name: Name,
+    pub span: Span,
 }
```

Comment:
Try modifying the `derive` to not include `PartialEq` and implement it yourself as such:

```rust
impl PartialEq for ParamTy {
    fn eq(&self, other: &Self) -> bool {
        (self.idx, self.name) == (other.idx, other.name)
    }
}
```

The compilation went past `compiler_builtins` with that change.